[AerollingOverDeepBlueSeas] iP

AerollingOverDeepBlueSeas opened the pull request:
# Tete
## On having a tête-à-tête
> With great wisdom come great loquaciousness 😉

- Unordered thing 1 
- Unordered thing 2 but
- 1. Ta-da
- 2. Potato Salad

```java
int i = 0;
String s;
```

- [x] ~Created PR~
- [ ] Complete **all** tasks
- [ ] *Dream in binary*

[Google](https://www.google.com)

Inline `code` has `back-ticks around` it. Like this: `Boogie Woogie`

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

AerollingOverDeepBlueSeas committed:
Week 2 Project (All Levels Except A-enums)

AerollingOverDeepBlueSeas committed:
A-Enums

AerollingOverDeepBlueSeas committed:
Create File to read.

AerollingOverDeepBlueSeas committed:
Complete level 7

AerollingOverDeepBlueSeas committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Complete level 7
  Create File to read.

AerollingOverDeepBlueSeas committed:
Complete Level 8

AerollingOverDeepBlueSeas committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Complete Level 8

AerollingOverDeepBlueSeas committed:
Add Storage and TaskList

AerollingOverDeepBlueSeas committed:
Complete A-MoreOOP requirements

AerollingOverDeepBlueSeas committed:
Merge branch 'more-OOP'

AerollingOverDeepBlueSeas committed:
Move contents into duke package

AerollingOverDeepBlueSeas committed:
Merge remote-tracking branch 'origin/add-gradle-support'

AerollingOverDeepBlueSeas committed:
Package class files

AerollingOverDeepBlueSeas committed:
Complete A-Gradle, A-JUnit and A-Jar

AerollingOverDeepBlueSeas committed:
Complete A-JavaDoc

AerollingOverDeepBlueSeas committed:
Complete Level-9

AerollingOverDeepBlueSeas committed:
Merge branch 'branch-A-JavaDoc'

AerollingOverDeepBlueSeas committed:
no message

AerollingOverDeepBlueSeas committed:
Merge branch 'branch-A-CodingStandard'

AerollingOverDeepBlueSeas committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Task.java

AerollingOverDeepBlueSeas committed:
Updated Level-9 so find is more flexible (description only needs to contain the search string, instead of exact match). Enforced OOP better by moving more display calls to UI.

RT0118 reviewed (commented):
Good work on Tete! It's great that the code mostly follows the proper naming convensions as described in the docs, but one thing has caught my attention. Within your task classes, perhaps the `boolean done` variable could be more intuitively named? Other than that, I like that most variables have proper naming. Also, good use of proper use of nouns and verbs (like the toData function name).

AerollingOverDeepBlueSeas committed:
Implement checkstyle

nanditaselvaraju commented on src/test/java/duke/TaskListTest.java:
```diff
@@ -0,0 +1,87 @@
+    }
+
+    @Test
+    public void unmarkItem_invalidIndexException_exceptionThrown() {
```
not sure if this is recommended, can try to keep the method names shorter

nanditaselvaraju commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,231 @@
+                
+                \tYou may want to run the command 'list' again.\
+                
+                \tFortunately, one of us here has the items remembered.""", e.getMessage());
```
I think it's good that your program has a bit of personality!

nanditaselvaraju commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,231 @@
+    }
+
+    @Test
+    public void addIncompleteEvent_f_missingFieldContentsException_exceptionThrownTest() {
```
same comment, make can keep the method names shorter, and add a comment about what it is testing instead

nanditaselvaraju reviewed (commented):
to my knowledge,  i think your program is good and well-written. just that the method names for the testing could be shorter

AerollingOverDeepBlueSeas committed:
Add initial GUI

AerollingOverDeepBlueSeas committed:
Complete level 10 requirements

AerollingOverDeepBlueSeas committed:
Merge branch 'branch-Level-10'

davidfoo07 commented on src/main/java/tete/TaskList.java:
```diff
@@ -0,0 +1,165 @@
+     *
+     * @param line containing contents of the task
+     */
+    public void addItemFromFile(String line) {
```
It is great that you are adding item from a file. One possible improvement is refactoring the method to handle more edge cases too. For instance, what happens if the file contains an invalid entry. Considering adding an error handling here to prevent the app from crashing.

davidfoo07 commented on src/main/java/tete/Task.java:
```diff
@@ -0,0 +1,80 @@
+     * Returns different string depending on completion status.
+     * @return X if task is complete, and an empty space otherwise.
+     */
+    public String getStatus() {
```
The getStatus() checks if a task is being done using a ternary operator. It is used in the toString() method. However, this can be simpliefied by directly formatting the toString() method to this.isDone ? "X" : " " instead of creating an additional getStatus() functions. This is applicable to every getStatus() functions in this program.

davidfoo07 commented on src/main/java/tete/TaskList.java:
```diff
@@ -0,0 +1,165 @@
+            int index = validateIndex(input);
+            System.out.println(tasks.size());
+            UI.displaySuccessfulDeleteMessage(tasks, tasks.remove(index));
+            return UI.returnSuccessfulDeleteMessage(tasks, tasks.remove(index));
```
Found duplicated tasks.remove(index), which is unnecessary. You refactor the code to first remove the task and then pass the result to the display method.

davidfoo07 commented on src/main/java/tete/InvalidCommandException.java:
```diff
@@ -0,0 +1,13 @@
+public class InvalidCommandException extends TeteException {
+
+    /** Creates a new InvalidCommandException. */
+    public InvalidCommandException() {
```
Good messages provided in each execption which guide the users in correcting their mistakes.

AerollingOverDeepBlueSeas committed:
Add assertions and update GUI

AerollingOverDeepBlueSeas committed:
Reduce line length for long lines

AerollingOverDeepBlueSeas committed:
Merge branch 'branch-A-Assertions'

AerollingOverDeepBlueSeas committed:
Merge branch 'branch-A-CodeQuality'

AerollingOverDeepBlueSeas committed:
Rename Duke to Tete

AerollingOverDeepBlueSeas committed:
Merge branch 'branch-A-Streams'

AerollingOverDeepBlueSeas committed:
Enable check of duplicate tasks

There is no check of duplicates when adding new tasks.

This causes issues when marking tasks as completed - it is not clear
whether they are the same task and accidentally duplicated or they are
multiple and distinct instances of the same task.

Adding this check would remove this ambiguity.

Let's add this check right before adding the new task to the list,
so that a response can be made to the user in lieu of a success
message.

AerollingOverDeepBlueSeas committed:
Merge branch 'C-DetectDuplicates'

AerollingOverDeepBlueSeas committed:
Refactor parser

Validation and execution of all commands are done under one method.

This results in deep nesting, and a method that is longer than it
needs to be.

Pulling out the validation and execution from the method to form their
own distinct methods allows for greater readability of code.

Let's pull out the validation and execution of each command into their
separate methods, and call them under the original parsing method.

Doing so makes each component easier to debug as the different methods
can be tested separately.

AerollingOverDeepBlueSeas committed:
Update README.md

AerollingOverDeepBlueSeas committed:
Merge pull request #2 from AerollingOverDeepBlueSeas/branch-Readme

Update README.md

AerollingOverDeepBlueSeas committed:
Tweak text to better fit GUI format

AerollingOverDeepBlueSeas committed:
Improve GUI and text formatting

AerollingOverDeepBlueSeas committed:
Add help command and update GUI

AerollingOverDeepBlueSeas committed:
Merge branch 'Branch-A-BetterGui'

AerollingOverDeepBlueSeas committed:
Update README.md

AerollingOverDeepBlueSeas committed:
Add UI Screenshot

AerollingOverDeepBlueSeas committed:
Refactor UI to Message for release

AerollingOverDeepBlueSeas committed:
Update README.md

AerollingOverDeepBlueSeas committed:
Update README.md
